servicetalk-grpc-netty/src/main/java/io/servicetalk/grpc/netty/DefaultGrpcServerBuilder.java

```diff
@@ -197,6 +206,11 @@ protected void doAppendHttpServiceFilter(final Predicate<StreamingHttpRequest> p
 
     private HttpServerBuilder preBuild() {
         if (!invokedBuild) {
+            // TODO(dj): Consider logging a warning in every method that changes the state of the builder once it has
+            // been used to build a client.
+            if (initializer != null) {
```

Comment:
It will be unexpected to users if they reuse the builder and change `HttpInitializer` between different builds to apply different socket options or filters. We should preserve the typical builder pattern.

Consider changing the ctor argument from `HttpServerBuilder` to `Supplier<HttpServerBuilder>`. Then you will be able to create a fresh new HTTP builder every time users try to build a gRPC server.
For deprecated methods, consider having a separate "deprecatedHttpInitializer" variable and compose a new `HttpInitializer` every time users invoke deprecated methods by appending a new configuration to the end. At build time, invoke `deprecatedHttpInitializer` first, then `httpInitializer`.